app/services/search_service.rb

```diff
@@ -184,6 +194,11 @@ def find_users(query, limit, type = nil)
       else
         User.where('username LIKE ? AND rusers.status = 1', '%' + query + '%')
       end
+
+    if user_tag.present?
+      users = User.joins(:user_tags)
+                  .where('user_tags.value LIKE ?', user_tag)
+                  .where(id: users.select("rusers.id"))
```

Comment:
Hm. @sagarpreet-chadha or @milaaraujo -- I'm seeing: 

`SyntaxError:         SyntaxError: /app/app/services/search_service.rb:206: syntax error, unexpected end-of-input, expecting keyword_end`

i.e. a missing `end` somewhere...

Unfortunately i can't see where this has happened! This block is my best guess, but when i count them, i don't see one missing. 

Can someone give a look? I think my brain is melting a bit on my first day back at work... sorry!

Reply:
I think the problem is in the `find_users()` method. This is what we have in the master branch:
```
  def find_users(query, limit, type = nil)
    users =
      if type == "tag"
        User.where('rusers.status = 1')
            .joins(:user_tags)\
            .where('user_tags.value LIKE ?', '%' + query + '%')\
      else if ActiveRecord::Base.connection.adapter_name == 'Mysql2'
        type == "username" ? User.search_by_username(query).where('rusers.status = ?', 1) : User.search(query).where('rusers.status = ?', 1)
      else
        User.where('username LIKE ? AND rusers.status = 1', '%' + query + '%')
      end
    end

    users = users.limit(limit)
  end
```
The merge process ended up merging the wrong thing... Try to replace the `find_users()` method with the code I copied above.

Reply:
And it worked! \o\ /o/